addons/content-browse/common/src/main/java/org/commonjava/indy/content/browse/ContentBrowseController.java

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright (C) 2013 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.content.browse;
+
+import org.commonjava.indy.IndyWorkflowException;
+import org.commonjava.indy.content.ContentManager;
+import org.commonjava.indy.content.StoreResource;
+import org.commonjava.indy.content.browse.model.ContentBrowseResult;
+import org.commonjava.indy.data.IndyDataException;
+import org.commonjava.indy.data.StoreDataManager;
+import org.commonjava.indy.model.core.ArtifactStore;
+import org.commonjava.indy.model.core.StoreKey;
+import org.commonjava.indy.model.core.StoreType;
+import org.commonjava.indy.model.galley.KeyedLocation;
+import org.commonjava.indy.util.ApplicationStatus;
+import org.commonjava.indy.util.UriFormatter;
+import org.commonjava.maven.atlas.ident.util.JoinString;
+import org.commonjava.maven.galley.event.EventMetadata;
+import org.commonjava.maven.galley.model.ConcreteResource;
+import org.commonjava.maven.galley.model.Transfer;
+import org.commonjava.maven.galley.model.TransferOperation;
+import org.commonjava.maven.galley.util.PathUtils;
+import org.commonjava.maven.galley.util.UrlUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import static org.apache.commons.lang.StringUtils.join;
+import static org.commonjava.maven.galley.util.PathUtils.normalize;
+import static org.commonjava.maven.galley.util.PathUtils.parentPath;
+
+@ApplicationScoped
+public class ContentBrowseController
+{
+    private final Logger logger = LoggerFactory.getLogger( this.getClass() );
+
+    @Inject
+    private StoreDataManager storeManager;
+
+    @Inject
+    private ContentManager contentManager;
+
+    protected ContentBrowseController()
+    {
+    }
+
+    public ContentBrowseController( final StoreDataManager storeManager, final ContentManager contentManager )
+    {
+        this.storeManager = storeManager;
+        this.contentManager = contentManager;
+    }
+
+    public ContentBrowseResult browseContent( final String packageType, final String type, final String name,
```

Comment:
Let's put StoreKey construction in the REST endpoint to cut down on parameters here. Then, we can do basic validation of user input at that level too, if we need to.